sql-plugin/src/main/scala/org/apache/spark/sql/rapids/aggregate/aggregateFunctions.scala

```diff
@@ -2023,3 +2023,143 @@ case class GpuVarianceSamp(child: Expression, nullOnDivideByZero: Boolean)
 
   override def prettyName: String = "var_samp"
 }
+
+case class GpuReplaceNullmask(
+    input: Expression, 
+    mask: Expression) extends GpuExpression with ShimExpression {
+
+  override def dataType: DataType = input.dataType
+  override def nullable: Boolean = mask.nullable
+  override def children: Seq[Expression] = Seq(input, mask)
+
+  override def columnarEval(batch: ColumnarBatch): GpuColumnVector = {
+    val res = withResource(input.columnarEval(batch)) { inputColumn =>
+      withResource(mask.columnarEval(batch)) { maskColumn =>
```

Comment:
maskCloumn can be closed right away after getting the maskColumnNotNull column to reduce te peak memory usage during executing.
```
val maskColumnNotNull = withResource(mask.columnarEval(batch)) { maskColumn =>
  maskColumn.getBase.isNotNull
}
withResource(maskColumnNotNull) { _ =>
  inputColumn.getBase.copyWithBooleanColumnAsValidity(maskColumnNotNull)
}
```

Reply:
done.